### ISSUE TITLE
[FEATURE_REQUEST] Poling the client every once in a while so Cloudflare won't timeout the session.

### ISSUE THREAD

**Issue by theman23290** [SillyTavern/SillyTavern#1925]
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

Cloudflare has a policy in place that will close any open connections without a response after 100 seconds resulting in an error 524. If Sillytavern can send a ping to the open session every 30 sec or so, Cloudflare will keep the connection open and allow for more than 100 second generation. This is a problem mainly when I want to generate a large image and the connection keeps getting closed because Sillytavern won't respond until the image is fully generated.

### Describe the solution you'd like

Put something in place that will ping the client every 30-70 seconds during generation or all the time so that Cloudflare or other similar services won't close the connection for inactivity.

### Describe alternatives you've considered

This is a hard limit set be Cloudflare and the only work around is paying them $200 a month to unlock that hard limit.

### Additional context

None

### Priority

Medium (Would be very useful)

### Is this something you would be keen to implement?

Maybe

**Comment by rosx27** [comment]
are you using cloudflare for free? (i think that's the one without custom domain name or something)

**Comment by theman23290** [comment]
I am using the paid Pro tier. That one still has the limit of 100 sec on it.

**Comment by rosx27** [comment]
i see. i was using a free tier, the one with a random domain provided to you everytime you open up a tunnel. but I got tired of frequently setting it up so I looked for a registered domain name, however, the lowest tier is kinda out of my budget haha

anyway, I'm currently using a free tier of tailscale and it does not have a time limit (tho it has a session limit ie, it only can serve up to 3 different google accounts (its fine in my case) and up to 100 devices (plenty enough for my 3 devices)). its slow though, but enough for the use of sillytavern.
